_posts/2022-04-12-cockpit-267.md

```diff
@@ -0,0 +1,120 @@
+
+![screenshot of install kpatches only]({{ site.baseurl }}/images/267-software-updates-install-kpatches-only.png)
+
+## metrics: Add CPU temperature to metrics page
+
+State: **OPEN** (will not be included) @ <https://github.com/cockpit-project/cockpit/pull/17160>
+{:.warn}
+
+Fixes: #11364
+
+![cpu_temperature_normal]({{ site.baseurl }}/images/267-metrics-add-cpu-temperature-to-metrics-page.png)
+
+![cpu_temperature_warning]({{ site.baseurl }}/images/267-metrics-add-cpu-temperature-to-metrics-page-2.png)
+
+![cpu_temperature_critical]({{ site.baseurl }}/images/267-metrics-add-cpu-temperature-to-metrics-page-3.png)
+
+## bridge: Add channel redirection
+
+State: **OPEN** (will not be included) @ <https://github.com/cockpit-project/cockpit/pull/14641>
+{:.warn}
+
+This adds a "redirect" option to the open command, which takes the name
+of the channel the output of the new one will be redirected to. The
+redirect is effectively applied as close to the channels as possible.
+
+Closes #14560
+
+---
+
+There are two cases in which the output is redirected from a channel:
+- The output is redirected from a local channel. This is handled in `CockpitRouter` by setting the `redirect_target` property of `CockpitChannel`, which uses it to send data to the redirect target instead of the transport in `cockpit_channel_actual_send`.
+- The output is redirected from a peer. This is handled in `cockpit_peer_handle` by passing the redirect target as an argument and then by putting the redirect into a hash table with the (source) channel name as a key. `on_other_recv` then looks up the channel name in the table, again sending it to the redirect target instead of the transport if a corresponding entry is present.
+
+Similarly there are two redirect target types: a local channel and a peer. These are handled by different implementations of the `CockpitRedirect` interface - `CockpitChannelRedirect` for local channels (holding a reference to the corresponding `CockpitChannel` object) and `CockpitPeerRedirect` for peers (holding a reference to the transport and the channel name).
+
+Redirection of a channel is attempted on all bridges - the shortest path then takes precedence (e.g. if there is a redirect between two channels on the same peer, it happens on both the parent bridge and the peer, but effectively the data is redirected on the peer and nothing arrives on the parent). This also means it is compatible with older versions of cockpit-bridge - they simply ignore the redirect option and the redirect is done on the parent bridge (this can obviously be slower, but it will work).
+
+I hope this makes sense - it seems like a small, but useful feature to me. (For potential uses see the linked issue.)
+
+## Machines: Remove static hosts from virtual network's DHCP
+
+State: **merged** @ <https://github.com/cockpit-project/cockpit-machines/pull/640>
+{:.note}
+
+The "static host entries" now have a "remove" action link:
```

Comment:
"action link" sounds very web developer lingo-y :)

Reply:
Good point -- changed to just "action" in https://github.com/cockpit-project/cockpit-machines/pull/640 (will pick up on next refresh, but that needs #515 to land)